SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialPlayerSpawner.cpp

```diff
@@ -298,20 +312,34 @@ void USpatialPlayerSpawner::ReceiveForwardedPlayerSpawnRequest(const Worker_Comm
 		return;
 	}
 
-	FUnrealObjectRef PlayerStartRef = GetObjectRefFromSchema(Payload, SpatialConstants::FORWARD_SPAWN_PLAYER_START_ACTOR_ID);
+	bool bRequestHandledSuccessfully = true;
 
-	bool bUnresolvedRef = false;
-	if (AActor* PlayerStart = Cast<AActor>(FUnrealObjectRef::ToObjectPtr(PlayerStartRef, NetDriver->PackageMap, bUnresolvedRef)))
+	FUnrealObjectRef PlayerStartRef = GetObjectRefFromSchema(Payload, SpatialConstants::FORWARD_SPAWN_PLAYER_START_ACTOR_ID);
+	if (PlayerStartRef != FUnrealObjectRef::NULL_OBJECT_REF)
 	{
-		UE_LOG(LogSpatialPlayerSpawner, Log, TEXT("Received ForwardPlayerSpawn request. Client worker ID: %s. PlayerStart: %s"), *ClientWorkerId, *PlayerStart->GetName());
+		bool bUnresolvedRef;
+		AActor* PlayerStart = Cast<AActor>(FUnrealObjectRef::ToObjectPtr(PlayerStartRef, NetDriver->PackageMap, bUnresolvedRef));
+		bRequestHandledSuccessfully = !bUnresolvedRef;
+
+		if (bRequestHandledSuccessfully)
+		{
+			UE_LOG(LogSpatialPlayerSpawner, Log, TEXT("Received ForwardPlayerSpawn request. Client worker ID: %s. PlayerStart: %s"), *ClientWorkerId, *PlayerStart->GetName());
+		}
+		else
+		{
+			UE_LOG(LogSpatialPlayerSpawner, Error, TEXT("PlayerStart Actor UnrealObjectRef was invalid on forwarded player spawn request worker: %s"), *ClientWorkerId);
+		}
+
 		PassSpawnRequestToNetDriver(PlayerSpawnData, PlayerStart);
 	}
 	else
 	{
-		UE_LOG(LogSpatialPlayerSpawner, Error, TEXT("PlayerStart Actor UnrealObjectRef was invalid on forwarded player spawn request worker: %s. Defaulting to normal player spawning flow."), *ClientWorkerId);
+		UE_LOG(LogSpatialPlayerSpawner, Log, TEXT("PlayerStart Actor was null object ref in forward spawn request. This is intentional when handing request to the correct "
+			"load balancing layer. Attempting to find a player start again."));
+		FindPlayerStartAndProcessPlayerSpawn(PlayerSpawnData, ClientWorkerId);
```

Comment:
Any thoughts on how we prevent an endless loop here, if this worker returns a player start on the original worker?

Reply:
Maybe `PassSpawnRequestToNetDriver(PlayerSpawnData, PlayerStart);` we call this here instead will nullptr for the player start?

Reply:
I don't think it will infinite loop. I think if this worker returns a PlayerStart on the original worker, this is received as an UnresolvedRef rather than a NullRef, and in that case we effectively are doing `PassSpawnRequestToNetDriver(PlayerSpawnData,  nullptr);`?

Reply:
Why would it be an UnresolvedRef?
Server 1 - chooses well known start point on server 2 - server 2 chooses start point well know start point on server 1. ?

Reply:
We only call `FindPlayerStartAndProcessPlayerSpawn` on Server 2  if Server 1 passes a `NULL_OBJECT_REF`. If Server 1 passes a `NULL_OBJECT_REF`, it's because it couldn't find any PlayerStarts, so deferred to the GameMode auth server. If you find a PlayerStart which is authoritative on Server 1 (which has no PlayerStarts), that's a critical error on the user end. Would you prefer some specific logic to make this error more explicit?

Reply:
Yeah the flow I was thinking that could cause it -
Server 1 - asked to pick player start, checks for auth on game mode and see it isn't, so returns nullptr
Server 2 - asked to pick player start, finds one, but it's auth on server 1 layer.
Definitely user error but I'm fine with this for now.